programs/rewards/src/instructions/withdraw_mining.rs

```diff
@@ -33,6 +35,22 @@ pub fn process_withdraw_mining<'a>(
 
     let delegate_mining = get_delegate_mining(delegate_mining, mining)?;
 
+    if let Some(delegate_mining) = delegate_mining {
```

Comment:
should we combine this check with get_delegate_mining? It's always called after it anyways

Reply:
I moved it to a separate function. It seems to me that `getting` and `verification` should be separate because if you merge them together it will make very logically overloaded function.

Reply:
But if you disagree I will merge it in one function

Reply:
I agree that it might be two separated functions without any problems.